### PR TITLE
Average Meeting Attendance

### PR DESCRIPTION
**Pull request by SnobbyDragon**
I'm sure there's a better way to do this that uses the joined tables feature, but I just counted everything. To test:
1. Comment out this line `ssl: {rejectUnauthorized: false}  // Comment out this line if connecting to local DB` in `postgres/pg-query.js`
2. Add some student ids to the student table
3. Add some meeting ids to the meeting table
4. Add student ids and meeting ids to the meeting_student table (beware that you cannot add ids that do not exist in the previous two tables)
5. Run `npm run dev`
6. Go to `http://localhost:3000/api/meeting/averageAttendance`

### CODE REVIEW

File: acmw-db-app/postgres/pg-query.js [hunk 1]
```diff
@@ -5,7 +5,7 @@ export default async (query) => {
     dotenv.config({ silent: process.env.NODE_ENV === 'production' });
     const dbConfig = {
         connectionString: process.env.DATABASE_URL,
-        ssl: {rejectUnauthorized: false}  // Comment out this line if connecting to local DB
+//        ssl: {rejectUnauthorized: false}  // Comment out this line if connecting to local DB
```
Comment (jgeorge37) [inline acmw-db-app/postgres/pg-query.js:8]: you need to un-comment this line before committing

File: acmw-db-app/pages/api/meeting/[pid].js [hunk 2]
```diff
@@ -0,0 +1,38 @@
+import pgQuery from '../../../postgres/pg-query.js';
+
+// GET /api/meeting/averageAttendance
+async function averageAttendance () {
+    var attendees = 0;
+    const meetingIds = await pgQuery(`SELECT id FROM meeting`);
```
Comment (jgeorge37) [inline acmw-db-app/pages/api/meeting/[pid].js:6]: this seems unnecessary, there should be a cleaner way

File: acmw-db-app/pages/api/meeting/[pid].js [hunk 3]
```diff
@@ -0,0 +1,38 @@
...
+    for (var i = 0; i < meetingIds.rows.length; i++) {
```
Comment (jgeorge37) [inline acmw-db-app/pages/api/meeting/[pid].js:7]: while the meeting_student table is theoretically cleared out after every academic year, i would prefer to explicitly check that the meetings you are counting are within the given academic year by checking the semester column values

**Review by jgeorge37 — CHANGES_REQUESTED** [review]

File: acmw-db-app/pages/api/meeting/[pid].js [hunk 4]
```diff
@@ -0,0 +1,53 @@
+import pgQuery from '../../../postgres/pg-query.js';
+
+// GET /api/meeting/averageAttendance
+async function averageAttendance () {
+    const d = new Date();
```
Comment (SnobbyDragon) [inline acmw-db-app/pages/api/meeting/[pid].js:5]: This is pretty redundant with the unique members per academic year. Maybe make a shared file that has commonly used functions like find current academic year?

Comment (jgeorge37) [reply acmw-db-app/pages/api/meeting/[pid].js:5]: yeah I can do that for the ticket I'm working on rn (ADA-73)

File: acmw-db-app/pages/api/meeting/[pid].js [hunk 5]
```diff
@@ -0,0 +1,53 @@
...
+    if (8 <= month <= 12) {
+      // it's currrently fall semester
+      fall += year;
+      spring += (year + 1);
+    } else {
+      //it's currrently spring (or summer)
+      fall += (year - 1);
+      spring += year;
+    }
+    const data = await pgQuery(`SELECT AVG(COUNT) FROM
```
Comment (jgeorge37) [inline acmw-db-app/pages/api/meeting/[pid].js:18]: would prefer to have less aggressive indentation here

File: acmw-db-app/pages/api/meeting/[pid].js [hunk 6]
```diff
@@ -0,0 +1,53 @@
+import pgQuery from '../../../postgres/pg-query.js';
+
+// GET /api/meeting/averageAttendance
+async function averageAttendance () {
+    const d = new Date();
+    const year = d.getFullYear() - 2000;
+    const month = d.getMonth();
+    var fall = "AU", spring = "SP";
```
Comment (jgeorge37) [inline acmw-db-app/pages/api/meeting/[pid].js:8]: "let" would be better than "var" since we only need block scope

**Review by jgeorge37 — CHANGES_REQUESTED** [review]
See comments on files. Additionally, when I try to get the average attendance if my meeting_student only has events outside of the current year, there is no response - I should get 0. Furthermore, if I have 3 meetings for AU20 in my meeting table and only two entries in meeting_student that are for the same meeting, I get 2 when I should get 0.6667 (2 people divided by 3 meetings).

**Review by jgeorge37 — CHANGES_REQUESTED** [review]
The new changes do not fix everything in the last review I left, and please see the comments I left about indentation and let vs var

**Review by jgeorge37 — APPROVED** [review]
i approve

**Pull request merged**